src/parser/meta.rs

```diff
@@ -47,17 +32,27 @@ pub fn parse_meta_get_status(buf: &[u8]) -> IResult<&[u8], Response> {
     ))(buf)
 }
 
-// TODO: refactor this into individual parse_meta_*_response methods
+pub fn parse_meta_get_response(buf: &[u8]) -> Result<Option<(usize, Response)>, ErrorKind> {
+    let bufn = buf.len();
+    let result = parse_meta_get_data_value(buf);
+
+    match result {
+        Ok((left, response)) => {
+            let n = bufn - left.len();
+            Ok(Some((n, response)))
+        }
+        Err(nom::Err::Incomplete(_)) => Ok(None),
+        Err(nom::Err::Error(e)) | Err(nom::Err::Failure(e)) => {
+            Err(ErrorKind::Protocol(Some(e.code.description().to_string())))
+        }
+    }
+}
+
 #[allow(dead_code)]
-pub fn parse_meta_response(buf: &[u8]) -> Result<Option<(usize, Response)>, ErrorKind> {
+pub fn parse_meta_set_response(buf: &[u8]) -> Result<Option<(usize, Response)>, ErrorKind> {
     let bufn = buf.len();
-    // TODO: alt calls parsers sequentially until one succeeds, which is not optimal.  Want to only call the correct parser, no sequencing.
-    // Also run the risk of hitting the wrong parser since there is overlap in response codes.
-    let result = alt((
-        parse_meta_status,
-        |input| parse_meta_get_data_value(input),
-        |input| parse_meta_set_data_value(input),
-    ))(buf);
```

Comment:
Removed because `alt` goes through unnecessary checks vs directly calling the intended method, and can also lead to unintended parsing behaviour (i.e. there are situations where a `meta_set` op could trigger `parse_meta_get_data_value` which has different logic & causes confusing behaviour)